src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Remote/RazorServices.cs

```diff
@@ -28,6 +28,7 @@ internal static class RazorServices
         [
             (typeof(IRemoteSignatureHelpService), null),
             (typeof(IRemoteInlayHintService), null),
+            (typeof(IRemoteDocumentSymbolService), null),
```

Comment:
>IRemoteDocumentSymbolService [](http://example.com/codeflow?start=20&length=28)

So we decided to go with Json serialization here because of the returned data complexity?

Reply:
Yeah, I decided it was easier to use Json for this one because the return from the endpoint is a SumType. The actual types involved, DocumentSymbol and SymbolInformation, are relatively simple though so I could be convinced to try wrapping things if people feel strongly about it.

Reply:
I don't have a problem at all :) just making sure I understood the reasoning (follow up from our conversation in the codeabout meeting :) )